cmd/metal-api/internal/metal/network.go

```diff
@@ -14,6 +14,7 @@ type MacAddress string
 type Nic struct {
 	MacAddress MacAddress `rethinkdb:"macAddress" json:"macAddress"`
 	Name       string     `rethinkdb:"name" json:"name"`
+	Alias      string     `rethinkdb:"alias" json:"alias"`
```

Comment:
Just a wild idea:

make alias a pointer of string, add to Nic a method `nic.ByIdentifier()` which internally checks for either Mac or Alias.

Then ByMac and ByAlias can be made private

Reply:
Does the metal API really have to know if the identifier is a MAC address or some weird string?

My gut feeling tells me that we should replace `MacAddress` simply by `Identifier`. For backward compatibility, we should introduce a new api version and ship the old and the new one several releases. Similar for the rethinkdb.

Reply:
Introducing new `Indentifier` field sounds like a good idea, yep.